### ISSUE TITLE
state_history_plugin maybe can not distinguish old and new delta

### ISSUE THREAD

**Issue by gzliudan** [EOSIO/eos#9255]
https://github.com/EOSIO/eos/blob/master/plugins/state_history_plugin/state_history_plugin.cpp#L532
```cpp
            for (auto& old : undo.old_values) {
               auto& row = index.get(old.first);
               if (include_delta(old.second, row))
                  delta.rows.obj.emplace_back(true, pack_row(row));
            }
            for (auto& old : undo.removed_values)
               delta.rows.obj.emplace_back(false, pack_row(old.second));
            for (auto id : undo.new_ids) {
               auto& row = index.get(id);
               delta.rows.obj.emplace_back(true, pack_row(row));
            }
```
The above codes use true for old records and new record, use false for removed records. But when there is no removed records,  we can't distinguish old records between new records, because all flags of records are true!

**Comment by swang-b1** [comment]
I think there is block number in the row delta update, which will indicate new and old.

**Comment by gzliudan** [comment]
I will investigate block number in the row delta update laster. But I think it's more clear that use a uint8_t to replace bool here, which value stards for 1=old; 0=removed; 2=new.

**Comment by gzliudan** [comment]
I decoded table delta data from state history plugin, no block number in row, so use bool can't distinguish between old values and new values.

> I think there is block number in the row delta update, which will indicate new and old.

**Comment by sanaraufx** [comment]
Should be fixed in 2.1
